ws/ws.go

```diff
@@ -99,7 +100,6 @@ func Write(key *datastore.Key, buf []byte) error {
 }
 
 func reader(ws *websocket.Conn) {
-	defer ws.Close()
```

Comment:
Well, that implies that the infinite loop breaks, which means there is an error returned on `ws.ReadMessage()`. Maybe that will tell us the root cause.